apprentice-follow-up/questions.md

```diff
@@ -6,11 +6,28 @@ Commit your answers to these questions in your copy
 Answer these after a day or two to see how much you remember
 
 1. What principle(s) can we apply to improve the size of images?
+- Use smaller base images (e.g., alpine)
+- Minimize the number of layers in Dockerfiles
+- Remove unnecessary files and dependencies
+
 1. What principle(s) can we apply to speed up rebuilds of images locally?
+- order the commands in the dockerfile so that the least changing commands come first. This allows us to leverage the caching mechanism better.
+- Reduce layers by combining related commands such as: RUN apt-get update && apt-get install
+
 1. What principles can we apply to speed up builds of an image in a pipeline?
+- Multi-stage builds
```

Comment:
In what way do you think Multi-stage builds will speed up builds in a pipeline?
What do you think the impact of the addition of copy commands between stages will be?